tensorflow/lite/delegates/gpu/metal_delegate.mm

```diff
@@ -211,7 +211,12 @@ kernel void ComputeFunction(device int* output_buffer [[buffer(0)]],
     for (auto& input : graph_inputs_) {
       if (input.tensor_id == tensor_index) {
         if (in_out_tensors_[input.id]->GetBufferHandle() != buffer) {
-          RETURN_IF_ERROR(in_out_tensors_[input.id]->SetBufferHandle(buffer));
+          if ([in_out_tensors_[input.id]->GetBufferHandle() length] == [buffer length]) {
+            RETURN_IF_ERROR(in_out_tensors_[input.id]->SetBufferHandle(buffer));
+            input.is_bphwc4 = true;
+          } else {
+            in_out_bhwc_f32_buffers_[input.id] = buffer;
```

Comment:
What's the use case of non phwc4 cases? How can we test it?

Reply:
This is the case when we use  `TFLGpuDelegateBindMetalBufferToTensor`. Like any other cpu inputs, user might bind a non-phwc4 buffer. If the user binds non-phwc4 buffer (just like any other cpu inputs), the original code won't work. The user has to always add a conversion code just like `TFLBufferConvert` . Note that in gl_delegate, they actually have a code that checks where it is pbwc4 or not for external buffer (https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/delegates/gpu/gl_delegate.cc#L195)

Reply:
I see. Could you add some comment for "in_out_bhwc_f32_buffers_[input.id] = buffer;" case ?